src/Raven.Server/Web/Studio/DataDirectoryInfo.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Http;
+using Raven.Server.Json;
+using Raven.Server.ServerWide;
+using Raven.Server.Utils;
+using Sparrow.Json;
+using Sparrow.Json.Parsing;
+using Sparrow.Logging;
+
+namespace Raven.Server.Web.Studio
+{
+    public class DataDirectoryInfo
+    {
+        private static readonly Logger Logger = LoggingSource.Instance.GetLogger<DataDirectoryInfo>("DataDirectoryInfo");
+
+        private readonly ServerStore _serverStore;
+        private readonly JsonOperationContext _context;
+
+        public DataDirectoryInfo(ServerStore serverStore, JsonOperationContext context)
+        {
+            _serverStore = serverStore;
+            _context = context;
+        }
+
+        public async Task<DataDirectoryResult> GetDatabaseDirectoryResult(
+            SingleNodeDataDirectoryResult currentNodeInfo, string path, string name)
+        {
+            var clusterTopology = _serverStore.GetClusterTopology();
+            var relevantNodes = clusterTopology.AllNodes.Keys;
+            var urlPath = $"admin/studio-tasks/full-data-directory?path={path}&name={name}";
+
+            var dataDirectoryResult = new DataDirectoryResult();
+            dataDirectoryResult.List.Add(currentNodeInfo);
+            
+            if (relevantNodes.Count > 1)
+            {
+                await UpdateNodesDirectoryResult(relevantNodes, clusterTopology, urlPath, dataDirectoryResult);
+            }
+
+            return dataDirectoryResult;
+        }
+
+        private async Task UpdateNodesDirectoryResult(
+            IEnumerable<string> nodes, ClusterTopology clusterTopology,
+            string urlPath, DataDirectoryResult dataDirectoryResult)
+        {
+            var httpClient = RequestExecutor.CreateHttpClient(_serverStore.Server.Certificate.Certificate, DocumentConventions.Default);
```

Comment:
Shouldn't this be disposed?

Reply:
The HttpClient is cached (by the certificate thumbprint)
https://github.com/ravendb/ravendb/blob/v4.1/src/Raven.Client/Http/RequestExecutor.cs#L132